### PR TITLE
SUS002 and SUS003 social case search/view

### PR DESCRIPTION
**Pull request by emilioward**
# Motivation and Context

Quite a big change be good to get opinions on this

This PR adds the first acceptance tests for the social stories searching for a postcode and displaying the case details for it 

# What has changed

Added social survey setup as part of tests and altered hardcoded default for business. These tests can be run without having to run make setup as a new survey is created prior to each feature.

New pages and steps

New make target for running social stories 

# How to test?

Run acceptance tests full suite

Test running just social stories `make social_acceptance_tests` (This will still setup business data as this is still part of the before all script. Further work needs to be done to decouple business from social in the tests)

# Links

SUS002 - https://digitaleq.atlassian.net/wiki/spaces/RASB/pages/639729701/SUS002+Search+by+postcode
SUS003 - https://digitaleq.atlassian.net/wiki/spaces/RASB/pages/639434773/SUS003+View+Case+Details

### CODE REVIEW

**Comment by emilioward** [comment]
[SUS003: View Case Details (8)](https://trello.com/c/oLmDvgmZ/188-sus003-view-case-details-8)

File: acceptance_tests/features/social_find_case_by_postcode.feature [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  @sus002-02
+  Scenario: User needs to enter a full postcode when searching
+    Given the SEL user receives a phone call from a respondent
+    When they search for a partial postcode
+    Then no postcodes/addresses are to be returned
+
+  @sus002-03
+  Scenario: The format of the postcode to be entered is XX11 X11 - supporting spaces and no spaces
+    Given the format of the postcode entered is XX11 X11
```
[the remark this answers is not in the document]
Comment (jcox-dev) [reply acceptance_tests/features/social_find_case_by_postcode.feature]: @benjefferies with r-ops running in a container? I thought the idea was to keep most of the UI tests within this repo.

File: controllers/sample_controller.py [hunk 2]
```diff
@@ -35,3 +35,23 @@ def upload_sample(collection_exercise_id, file_path):
     logger.info('Successfully uploaded sample file',
                 collection_exercise_id=collection_exercise_id, sample_file=file_path)
     return response_json
+
+
+def upload_social_sample(collection_exercise_id, file_path):
```
Comment (damien-lloyd) [inline controllers/sample_controller.py]: This function is very similar to the upload_sample function. Please merge the two and parameterise the function instead.

File: acceptance_tests/features/social_find_case_by_postcode.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (benjefferies) [reply acceptance_tests/features/social_find_case_by_postcode.feature]: With the two scenarios I've highlighted I don't think there is any extra UI logic being tested that's why I thought the sample service could be the most isolated place to test this

File: acceptance_tests/features/social_find_case_by_postcode.feature [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    And a social survey exists
+
+  @sus002-01
+  Scenario: User needs to be able to search for a case by postcode
+    Given the SEL user receives a phone call from a respondent
+    When they enter a postcode
+    Then all address are returned for the postcode
+
+  @sus002-02
+  Scenario: User needs to enter a full postcode when searching
```
[the remark this answers is not in the document]
Comment (emilioward) [reply acceptance_tests/features/social_find_case_by_postcode.feature]: Addressed in https://github.com/ONSdigital/rm-sample-service/pull/71

File: acceptance_tests/features/social_find_case_by_postcode.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (emilioward) [reply acceptance_tests/features/social_find_case_by_postcode.feature]: Addressed in https://github.com/ONSdigital/rm-sample-service/pull/71

File: acceptance_tests/features/social_find_case_by_postcode.feature [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    Then no postcodes/addresses are to be returned
+
+  @sus002-03
+  Scenario: The format of the postcode to be entered is XX11 X11 - supporting spaces and no spaces
+    Given the format of the postcode entered is XX11 X11
+    When the user clicks search
+    Then all address are returned for the postcode
+
+  @sus002-04
+  Scenario: The format of the postcode to be entered is XX11X11 - supporting spaces and no spaces
```
[the remark this answers is not in the document]
Comment (emilioward) [reply acceptance_tests/features/social_find_case_by_postcode.feature]: Addressed in https://github.com/ONSdigital/rm-sample-service/pull/71

**Pull request merged**